core/src/main/scala/io/gearpump/security/PasswordUtil.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gearpump.security
+
+import java.security.MessageDigest
+import sun.misc.{BASE64Decoder, BASE64Encoder}
+import scala.util.Try
+
+/**
+ * Util to verify whether user input password is valid or not.
+ * It use sha1 to do the digesting.
+ */
+object PasswordUtil {
+  private val SALT_LENGTH = 8
+
+  /**
+   * verification user input password with stored digest:
+   * base64Decode -> extract salt -> do sha1(salt, password) ->
+   * generate digest: salt + sha1 -> compare the generated digest with the stored digest.
+   */
+  def verify(password: String, stored: String): Boolean = {
+    Try {
+      val decoded = base64Decode(stored)
```

Comment:
Is there better decoder to use other than sun.misc - will this decoder exist if someone uses openjdk?